core/src/main/scala/kafka/server/KafkaApisUtils.scala

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import org.apache.kafka.common.record.Records
+import org.apache.kafka.common.requests.FetchResponse
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * the suites of helpers for KafkaApis class.
+ * We don't use companion object since KafkaApis is too fat.
+ */
+object KafkaApisUtils {
```

Comment:
I don't think creating a `KafkaApisUtils` makes things any better since it's unclear when to use this and when to use `KafkaApis`. If there is no utilities class related to quotas and requests handling, I would put it in the `KafkaApis` companion object.

Reply:
make sense. will replace utils by companion object.